core/src/main/kotlin/com/strumenta/kolasu/model/Indexing.kt

```diff
@@ -12,6 +12,16 @@ class SequentialIdProvider(private var counter: Int = 0) : IdProvider {
     }
 }
 
+class OnlyReferencedIdProvider(
+    private var idProvider: IdProvider = SequentialIdProvider()
+) : IdProvider {
+    override fun getId(node: Node): String? {
+        return node.properties
```

Comment:
Doesn't this only select nodes that refer to themselves?

Reply:
I think the work to understand which nodes are referred to should be done by who creates the value for withIds before calling generateJson. They can do that traversing the whole AST

Reply:
> Doesn't this only select nodes that refer to themselves?

You're right, my mistake, should be fixed now.

> I think the work to understand which nodes are referred to should be done by who creates the value for withIds before calling generateJson. They can do that traversing the whole AST

Done. The `generateJson` call now invokes `computeIdsForReferencedNodes`, which computes ids using the (no more default) ReferencedOnlyIdProvider id provider.